scipy/optimize/cython_optimize/__init__.py

```diff
@@ -0,0 +1,630 @@
+           cdef test_params myargs = args  # Cython automatically casts dictionary to struct
+           return coz.zeros_struct.brentq(f, xa, xb, <test_params *> &myargs, xtol, rtol, mitr, NULL)
+
+
+       def brentq_example(args=ARGS, xa=0.5, xb=1.0, xtol=XTOL, rtol=RTOL, mitr=MITR):
+           return brentq_wrapper_example(args, xa, xb, xtol, rtol, mitr)
+
+
+       x = brentq_example()
+       # 0.6999942848231314
+
+4. If you want to call your function from Python, create a wrapper
+5. Create a Cython ``.pxd`` file if you need to export any Cython functions
+
+Full Output
+-----------
+The  functions in :py:mod:`~scipy.optimize.cython_optimize` can also copy the
+full output from the solver to a C ``struct`` that is passed as its last
+argument. If you don't want the full output just pass ``NULL``. The full output
+``struct`` should contain the following:
+
+- ``funcalls``: number of function calls
+- ``iterations``: number of iterations
+- ``error_num``: error number
+
+The error number returned is -1 for a sign error, and -2 for a convergence
+error. All other values mean the solver converged. Note that the full output
+``struct`` must be cast to ``scipy_newton_parameters`` for the three functions
+in ``optimize/cython_optimize/Newton`` and ``scipy_zeros_parameters`` for the
+four functions in ``optimize/Zeros``. ::
+
+    from scipy.optimize.cython_optimize cimport zeros_tuple
+    from scipy.optimize.cython_optimize.examples.zeros_tuple_examples cimport f_solarcell, fprime, fprime2
+
+    ARGS = (5.25, 6.0, 1e-09, 0.004, 10.0, 0.27456)
+    XTOL, RTOL, MITR = 0.001, 0.001, 10
+
+    ctypedef struct scipy_brent_full_output:
+        int funcalls
+        int iterations
+        int error_num
+        double root
+
+    # cython brentq solver with full output
+    cdef scipy_brent_full_output solarcell_brent_full_output(tuple args, double xa, double xb,
+                                                             double xtol, double rtol, int mitr):
+        cdef scipy_brent_full_output full_output
+        full_output.root = zeros_tuple.brentq(f_solarcell, xa, xb, args,
+                                              xtol, rtol, mitr,
+                                              <zeros_tuple.scipy_zeros_parameters *> &full_output)
+        return full_output
+
+
+    def test_brent_full_output(args=ARGS, xa=0.0, xb=6.0, xtol=XTOL, rtol=RTOL, mitr=MITR):
+        '''test newton with full output'''
+        return solarcell_brent_full_output(args, xa, xb, xtol, rtol, mitr)
+
+    result = test_brent_full_output()
+    # {'funcalls': 4,
+    #  'iterations': 3,
+    #  'error_num': 281790720,
+    #  'root': 5.255231961257658}
+"""
+
+from __future__ import division, print_function, absolute_import
+
+__all__ = ['zeros_tuple', 'zeros_struct', 'zeros_array']
+
+from . import zeros_tuple, zeros_struct, zeros_array
+
+from scipy._lib._testutils import PytestTester
+test = PytestTester(__name__)
+del PytestTester
```

Comment:
better leave this out, not needed. `cython_blas/lapack/special` don't have this either.

Reply:
actually those modules don't have any `__init__.py`, they're just `.so`'s. why not do that here?